src/main/java/techcourse/myblog/repository/UserRepository.java

```diff
@@ -0,0 +1,22 @@
+package techcourse.myblog.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
+import techcourse.myblog.model.User;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("select u from User u where u.email = ?1")
+    User findUserByEmailAddress(String email);
+
+    @Transactional
+    @Modifying
+    @Query("delete from User u where u.email = ?1")
+    void deleteUserByEmailAddress(String email);
```

Comment:
아래의 메서드 이름을 사용하면 `@Query`는 필요 없을 거예요.

```suggestion
    void removeByEmail(String email);
```

Reply:
넵!

Reply:
위 코드에서
 ```java
void removeByEmail(String email); 
```
라고 수정했습니다.

다음과 같은 에러가 떴습니다.

No EntityManager with actual transaction available for current thread - cannot reliably process 
'remove' call 

위 코드에서 @Transactional을 추가했습니다.
 ```java
@Transactional
void removeByEmail(String email); 
```
테스트를 모두 통과하는 것을 확인했습니다.

왜 이렇게 되는지는 잘 모르겠습니다. 더 공부해 봐야할 것 같아요!